### DISCUSSION TITLE
Torch No Grad

### DISCUSSION THREAD

**Discussion by ghost** [deepjavalibrary/djl#1395] [Q&A]
I am in the process of rewriting my DRL code in Java (from Python) using the PyTorch engine.
Please can I be advised how create a torch.no_grad() context as in;

with torch.no_grad()
   etc, etc

Many thanks in anticipation.

**Comment by frankfliu** [comment]
@BobKelly1212 
If you are running inference, by default NDAarray we are not calculate grad.
Would you please provide more detail about what you are trying to achieve.

**Comment by saudet** [comment]
If you're looking for an API that is close to the Python of PyTorch, you're probably going to be more satisfied with the [JavaCPP Presets for PyTorch](https://github.com/bytedeco/javacpp-presets/tree/master/pytorch), which wraps the C++ API of PyTorch, and does have things like "no_grad", see https://github.com/bytedeco/javacpp-presets/issues/623#issuecomment-844749578.
